### PR TITLE
Intl.Locale updates

### PR DESCRIPTION
**Pull request by jahzielv**
A checklist to help your pull request get merged faster:
- [x] Summarize your changes
- [x] Data: link to resources that verify support information (such as browser's docs, changelogs, source control, bug trackers, and tests)
- [x] Data: if you tested something, describe how you tested with details like browser and version
- [x] Review the results of the linter and fix problems reported (If you need help, please ask in a comment!)
- [x] Link to related issues or pull requests, if any

This PR aims to add browser compat data for `Intl.Locale` and its various properties. This is part of a documentation effort involving Daniel Ehrenberg, Romulo Cintra, and myself. We're tracking our progress [here](https://github.com/tc39/proposal-intl-locale/issues/70). 

`npm test` runs cleanly and Travis builds successfully.

According to my research, Chrome (desktop, Android, and WebView) is the only browser with support currently ([shipped in Chrome 74](https://www.chromestatus.com/feature/4936310187884544)). [Firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=1433303) and [Edge](https://github.com/Microsoft/ChakraCore/pull/5675) have issues open and are in development, but have not implemented yet. [Safari](https://bugs.webkit.org/buglist.cgi?quicksearch=Intl.Locale) hasn't issued any public signals regarding `Intl.Locale`. 

## Properties of `Intl.Locale` to add compat data for:

- [x] prototype
- [x] minimize()
- [x] maximize()
- [x] toString()
- [x] baseName 
- [x] calendar 
- [x] collation 
- [x] hourCycle 
- [x] caseFirst 
- [x] numeric 
- [x] numberingSystem 
- [x] language
- [x] script
- [x] region
Cheers!

### CODE REVIEW

**Comment by jahzielv** [comment]
Edit note: I added a list of all the properties of `Intl.Locale` and whether or not I've added compat data for them. I'll be adding data over the coming days!

**Comment by vinyldarkscratch** [comment]
All the data is looking good to me!  A little pro tip with Chromium browsers: since Opera 15 (and Opera Android 14), Opera adapted Chrome's engine into its own.  For Opera desktop, the version numbers can be identified based upon the following formula: `min(15, CHROME_VERSION - 13)`.  (Opera Android was like that until version 43 -- I've [included a mapping in my pull request to add engine versions](https://github.com/mdn/browser-compat-data/pull/3877/files#diff-76b9c645f49a7fb342536bce7e613806)).  Samsung Internet's also Chrome-based, but for now, we just set it to `true` or `false`.

Regarding all the properties, would you like to add them in a single PR and wait for us to merge it until it's all done, or would you rather us merge this as it is and work on additional PRs?

**Comment by jahzielv** [comment]
Thanks for the tips @vinyldarkscratch! As far as the properties go, I think I'll add them all on this PR and then we can merge them all at once. I'll be offline this week, but should have some time Saturday to get back on this. Thanks!

**Comment by vinyldarkscratch** [comment]
Alright, sounds like a plan!  I’ll mark this as “not ready” for the time being.  😉

**Comment by jahzielv** [comment]
@vinyldarkscratch regarding your tip about Opera: Taking a look at this [version history](https://help.opera.com/en/opera-version-history/), it seems like the latest version (Opera 60) uses Chromium 73, which wouldn't support `Intl.Locale`. Could you elaborate a bit on what you meant by

> For Opera desktop, the version numbers can be identified based upon the following formula: min(15, CHROME_VERSION - 13). (Opera Android was like that until version 43 -- I've included a mapping in my pull request to add engine versions). Samsung Internet's also Chrome-based, but for now, we just set it to true or false.

? Thanks for your help with this!

**Comment by vinyldarkscratch** [comment]
Sorry, you can totally ignore that part!  I wrote that before I found out Opera 61 and 62 are only beta versions.  😛

**Comment by jahzielv** [comment]
Cool cool thanks!

**Comment by jahzielv** [comment]
All data has been added! `npm test` runs cleanly, but I have a question about the rendering. When I run a render test, the resulting table looks like this:

Screen Shot 2019-05-18 at 12 03 15 AM

The data is correct, but the table isn't styled correctly. Am I missing something in the command? I ran 

```bash
npm run render javascript.builtins.Intl.Locale.script
```

**Comment by vinyldarkscratch** [comment]
Don't mind the old styling with `npm run render`, efforts to get it to use the new styling sheet are in #736 and #737.  Thanks for your hard work on this PR!  I'll be reviewing it shortly.  :wink:

File: javascript/builtins/Intl.json [hunk 1]
```diff
@@ -1263,6 +1263,831 @@
...
+                "experimental": false,
+                "standard_track": true
+              }
+            }
+          },
+          "hourCycle": {
+            "__compat": {
+              "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Locale/hourCycle",
+              "spec_url": "https://tc39.github.io/proposal-intl-locale/#sec-Intl.Locale.prototype.hourCycle",
+              "status": {
```
Comment (vinyldarkscratch) [inline javascript/builtins/Intl.json]: It seems that starting here, `status` was put above `support`.  Can we reverse these?  (I'll only comment on this one.)

Comment (jahzielv) [reply javascript/builtins/Intl.json]: Sounds good! I'll make the switch 😄

**Comment by jahzielv** [comment]
Order has been updated! 🎉

**Comment by jahzielv** [comment]
Woot woot - thanks for your help!

**Pull request merged**